pkg/credentialstores/common.go

```diff
@@ -0,0 +1,129 @@
+package credentialstores
+
+import (
+	"net/url"
+	"strings"
+
+	"github.com/openshift/image-registry/pkg/kubernetes-common/credentialprovider"
+
+	"github.com/golang/glog"
+)
+
+// MultiError is a wrap for multiple errors.
+type MultiError struct {
+	errors []string
+}
+
+// Append adds a new error.
+func (m *MultiError) Append(e error) {
+	m.errors = append(m.errors, e.Error())
+}
+
+// Error exists to comply with golang's error interface.
+func (m *MultiError) Error() string {
+	return strings.Join(m.errors, " ")
+}
+
+// basicCredentialsFromKeyring extract basicAuth information from provided
+// keyring. If keyring does not contain information for the provided URL, empty
+// strings are returned instead.
+func basicCredentialsFromKeyring(keyring credentialprovider.DockerKeyring, target *url.URL) (string, string) {
+	regURL := getURLForLookup(target)
+	if configs, found := keyring.Lookup(regURL); found {
+		glog.V(5).Infof(
+			"Found secret to match %s (%s): %s",
+			target, regURL, configs[0].ServerAddress,
+		)
+		return configs[0].Username, configs[0].Password
+	}
+
+	// do a special case check for docker.io to match historical lookups
+	// when we respond to a challenge
+	if regURL == "auth.docker.io/token" {
```

Comment:
We don't support it

Reply:
I don't think we can change this as this is not something like a StaticCredential.

Reply:
Why is this not something like a StaticCredential?

Reply:
Registry client in library-go is designed around a Context entity. This entity is capable of, by means of a [CredentialStore](https://github.com/openshift/library-go/blob/master/pkg/image/registryclient/client.go#L57), fetch "repositories". 

Image registry uses library-go and creates a single CredentialStore containing all credentials available, that is IMO much better than creating individual Contexts with different "StaticCredentials" every time we need a Context to access a repository.